src/extension.ts

```diff
@@ -65,13 +70,20 @@ function activateTOC(
   // Add the ToC widget to the application restorer:
   restorer.add(toc, 'juputerlab-toc');
 
-  // Create a notebook generator:
-  const notebookGenerator = createNotebookGenerator(
-    notebookTracker,
-    toc,
-    rendermime.sanitizer
-  );
-  registry.add(notebookGenerator);
+  // Load the configuration of whether to enable collapsing behavior
+  settingRegistry.load('@jupyterlab/toc:plugin').then(settings => {
+    const config = settings.get('tocConfig').composite as JSONObject;
+    const collapsibleNotebooks = config.collapsibleNotebooks as boolean;
+
+    // Create a notebook generator:
+    const notebookGenerator = createNotebookGenerator(
+      notebookTracker,
+      toc,
+      collapsibleNotebooks,
```

Comment:
Depending on how you do this, as settings may fail to load, once extracted to parent scope, this can be `collapsibleNotebooks || false` in order to ensure that a boolean is passed along to the generator. In fact, once the settings schema is updated, you can elide a new variable declaration altogether and just do

```javascript
...,
toc,
settings.composite.collapsibleNotebooks as boolean,
...
```